File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;

        int count[26];
        memset(count,0,sizeof(count));
        
        for (int i = 0; i < s.size(); i++) {
            count[s[i] - 'a']++;
        }
        
        for (auto c:t) {
            --count[c - 'a'];
            if (count[c - 'a'] < 0) 
                return false;
        }

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut char_counts = [0; 26]; // Initialize array for character counts

        for c in s.chars() {
            char_counts[c as usize - 97] += 1; // Increment count for current char
        }

        for c in t.chars() {
            char_counts[c as usize - 97] -= 1; // Decrement count for characters in t
            if char_counts[c as usize - 97] < 0 {
                return false;
            }
        }

        true
    }
}